File: TA-PA2/mobile/kia_app/lib/features/edukasi/data/models/edukasi_trimester_model.dart
class EdukasiTrimesterModel {
  final int id;
  final String trimester;
  final String kategori;
  final String judul;
  final String isi;
  final String gambarUrl;

  EdukasiTrimesterModel({
    required this.id,
    required this.trimester,
    required this.kategori,
    required this.judul,
    required this.isi,
    required this.gambarUrl,
  });

  factory EdukasiTrimesterModel.fromJson(
    Map<String, dynamic> json,
  ) {
    return EdukasiTrimesterModel(
      id: int.parse(json['id'].toString()),
      trimester: json['trimester'] ?? '',
      kategori: json['kategori'] ?? '',
      judul: json['judul'] ?? '',
      isi: json['isi'] ?? '',
      gambarUrl: json['gambar_url'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'trimester': trimester,
      'kategori': kategori,
      'judul': judul,
      'isi': isi,
      'gambar_url': gambarUrl,
    };
  }
}

File: backend/monolith/go-template-main/app/models/edukasi_trimester_model.go
package models

import "time"

type EdukasiTrimester struct {
	ID         int32     `json:"id" gorm:"primaryKey;autoIncrement"`
	Trimester  string    `json:"trimester" gorm:"type:varchar(10)"`
	Kategori   string    `json:"kategori" gorm:"type:varchar(100)"`
	Judul      string    `json:"judul" gorm:"type:varchar(255)"`
	Isi        string    `json:"isi" gorm:"type:text"`
	GambarURL  string    `json:"gambar_url" gorm:"type:text"`
	CreatedAt  time.Time `json:"created_at" gorm:"->"`
	UpdatedAt  time.Time `json:"updated_at" gorm:"->"`
}

func (EdukasiTrimester) TableName() string {
	return "edukasi_trimester"
}